Analysis/Tasks/trackselection.cxx

```diff
@@ -0,0 +1,250 @@
+  OutputObj<TH1F> chi2PerClusterITS{TH1F(
+      "chi2PerClusterITS", "chi2PerClusterITS;chi2/cluster ITS", 100, 0, 50)};
+  OutputObj<TH1F> itsHits{TH1F("its-hits", "hitmap its", 8, -0.5, 7.5)};
+
+  // TPC related quantities
+  OutputObj<TH1F> tpcFindableClusters{TH1F(
+      "tpc-findableClusters", "Findable Clusters;;# tracks", 165, -0.5, 164.5)};
+  OutputObj<TH1F> tpcFoundClusters{
+      TH1F("tpc-foundClusters", "Found Clusters;;# tracks", 165, -0.5, 164.5)};
+  OutputObj<TH1F> tpcSharedClusters{TH1F(
+      "tpc-sharedClusters", "Shared Clusters;;# tracks", 165, -0.5, 164.5)};
+  OutputObj<TH1F> tpcFractionSharedClusters{
+      TH1F("tpc-fractionSharedClusters",
+           "Fraction of Shared Clusters;;# tracks", 500, -1.5, 1.5)};
+  OutputObj<TH1F> tpcCrossedRows{
+      TH1F("tpc-crossedRows", "Crossed Rows;;# tracks", 165, -0.5, 164.5)};
+  OutputObj<TH1F> tpcCrossedRowsOverFindableClusters{
+      TH1F("tpc-crossedRowsOverFindableClusters",
+           "Crossed rows over findable clusters;;# tracks", 110, 0.0, 1.1)};
+  OutputObj<TH1F> tpcChi2PerCluster{
+      TH1F("tpc-chi2PerCluster", "chi2 per cluster in TPC;chi2/cluster TPC",
+           100, 0, 10)};
+
+  // physics quantities
+  OutputObj<TH1F> pt{TH1F("track-pt", "pt", 100, 0., 50.)};
+  OutputObj<TH1F> eta{TH1F("track-eta", "eta", 102, -2.01, 2.01)};
+  OutputObj<TH1F> phi{TH1F("track-phi", "phi", 100, 0., 2 * M_PI)};
+
+  // track parameters
+  OutputObj<TH1F> x{TH1F(
+      "trackpar-x", "Track X at dca in local coordinate system;x [cm];# tracks",
+      500, -0.41, 0.41)};
+  OutputObj<TH1F> y{TH1F(
+      "trackpar-y", "Track Y at dca in local coordinate system;y [cm];# tracks",
+      100, -4., 4.)};
+  OutputObj<TH1F> z{TH1F(
+      "trackpar-z", "Track Z at dca in local coordinate system;z [cm];# tracks",
+      100, -20., 20.)};
+  OutputObj<TH1F> signed1Pt{TH1F("trackpar-signed1Pt",
+                                 "Track signed 1/p_{T};q/p_{T};# tracks", 1000,
+                                 -10, 10)};
+  OutputObj<TH1F> snp{TH1F(
+      "trackpar-snp", "Sinus of track momentum azimuthal angle;snp;# tracks",
+      1000, -0.0001, 0.0001)};
+  OutputObj<TH1F> tgl{TH1F(
+      "trackpar-tgl", "Tangent of the track momentum dip angle;tgl;# tracks",
+      1000, -2, 2)};
+
+  OutputObj<TH2F> collisionPos{
+      TH2F("collision-xy", "Position of collision;x [cm];y [cm];# tracks", 100,
+           -0.5, 0.5, 100, -0.5, 0.5)};
+
+  // for each filterbit separateHistos are required!
+
+  void init(o2::framework::InitContext &) {}
+
+  void process(aod::Collision const &collision,
+               soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov,
+                         aod::TrackFilterTable> const &tracks) {
+
+    collisionPos->Fill(collision.posX(), collision.posY());
+
+    for (auto &track : tracks) {
+
+      if (!track.isGlobalTrack())
+        continue; // FIXME: this should be replaced by framework filter
+
```

Comment:
And a configurable. In this way we can run the task once before and once after the selection